text/0095-nft-standard.md

```diff
@@ -0,0 +1,375 @@
+
+### Transferring an NFT
+
+- **Safe vs. Unsafe Transfers**: Use `safe_transfer_from` when transferring to a contract (to allow the contract to confirm it can handle NFTs) and `transfer_from` for basic transfers.
+- **Event Emission**: Both methods should update the owner mapping and emit a `Transfer` event.
+
+### Approval Mechanisms
+
+- **Per-token Approval**: The `approve` function enables an account or contract to transfer a specific NFT. If needed, revoke_approval clears that approval.
+- **Operator Approvals**: The `approve_for_all` and `revoke_approval_for_all` functions allow a user to delegate transfer rights over their entire collection, with an event emitted to signal the change.
+
+
+## Backend Indexing and NFT Metadata Tracking
+
+[backend-indexing-and-nft-metadata-tracking]: #backend-indexing-and-nft-metadata-tracking
+
+### Tracking NFT Events
+
+Backend indexers or off-chain processors should listen for NFT-related events (`CEP95Event` event enum):
+
+- **Minting**: Detect a `Mint` event. Extract the `token_id` and `to` address.
+- **Transfers and Approvals**: Capture changes in ownership or approvals by parsing `Transfer`, `Approval`, `RevokeApproval`, `ApprovalForAll` and `RevokeApprovalForAll` events.
+- **Burning**: Detect a `Burn` event. Extract the `token_id` and `from` address.
+
+### Tracking optional NFT Onchain Metadata
+
+Backend indexers or off-chain processors should listen for NFT Metadata related events (`CEP95MetadataEvent` event enum):
+
+1. Detect `MetadataUpdate` event.
+2. Extract the `token_id` and `metadata` from the event.
+3. Store the NFT’s `token_id` and `metadata` in your database for efficient querying and display in applications
+
+### Tracking optional NFT Offchain Metadata
+
+Once an event (such as minting) is detected:
+
+1. **Extract `token_id`**: From the event, identify the newly minted NFT. 
+2. **Query token URI**: Invoke or query the contract’s `token_uri(token_id)` function using Casper’s JSON-RPC methods (e.g., `state_get_item` or `state_get_dictionary_item`) or a read-only contract call. 
+3. **Fetch Off-Chain Metadata**: With the obtained URI, perform an HTTP/IPFS fetch to retrieve the JSON metadata. Parse the fields (name, description, asset_uri, etc.). 
+4. **Indexing**: Store the NFT’s `token_id`, `owner`, and `metadata` in your database for efficient querying and display in applications.
+
+## Conclusion
+
+[conclusion]: #conclusion
+
+The Casper NFT Standard (inspired by ERC-721) aims to bring uniformity and interoperability to NFT development on Casper. It provides simple methods for handling tokens, approvals, and optional metadata, making it easier for wallets, marketplaces, and indexers to work with NFTs. This proposal helps build a connected and thriving NFT ecosystem on Casper.
+
+## Appendix
+
+[appendix]: #appendix
+
+### Changelog
+
+- v1.0: Initial draft including CEP-95 and CEP95Metadata interfaces, event structures, and backend indexing guidelines.
```

Comment:
It's `CEP95` here. Let's use `code` formating for interfaces and plain text for standard names